Loading to spinner #2720

rtman opened the pull request:
### Description:

#2720 Updated lingering text based loading indicators. Except in the situation where the indicator is inside a button, nav component or otherwise needs to be constrained by a container. The current `LoadSpinner` component is a full screen component. I think the rest of those text based loading indicators could be swapped out for a similar component that isn't full screen but would fit to the parent component.

### Checklist:

- [✅] Test your work and double-check to confirm that you didn't break anything
- [✅] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [✅] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [✅] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

micahalcorn commented:
Thanks @rtman 🙏 Will you upload this branch as a [test build](https://github.com/OriginProtocol/test-builds) so that I can run it without jumping through a bunch of hoops?

I was hoping to get those contained "Loading..."s swapped out too, but that can be a different PR if this one gets us some easy wins.

micahalcorn commented on dapps/marketplace/src/components/Reviews.js:
```diff
@@ -184,7 +186,7 @@ export default class Reviews extends Component {
                   }}
                 >
                   {loading ? (
-                    <fbt desc="reviews.loadingMore">Loading More...</fbt>
```
It may be appropriate to run the translation commands to remove these now unused string instances. 🤔

rtman replied:
Ok I'll look into that.

rtman replied:
I've updated the translations

micahalcorn commented on dapps/marketplace/src/components/Reviews.js:
```diff
@@ -13,6 +13,8 @@ import QueryError from 'components/QueryError'
 import query from 'queries/Reviews'
 import EthAddress from './EthAddress'
 
+import LoadingSpinner from 'components/LoadingSpinner'
```
It may just be me, but I like to keep all the sibling `import`s grouped together, aka this one on line 11, perhaps. 😁

rtman replied:
Definitely not just you, I'll fix that!

rtman commented:
@micahalcorn I've uploaded a [test build](https://originprotocol.github.io/test-builds/loading-to-spinner) of the branch. Lmk if there's any issues.

And ya I think a separate PR for the remaining Loading indicators that this `LoadingSpinner` component won't work for is a good idea.  It'd have to not be `position: fixed` and be bound by the parent container.

rtman commented:
I notice that sometimes the loading spinner overlays the content and it doesn't look quite right. It seems to only be on the details page of a listing as far as I can tell. I'll have a look at why that is

micahalcorn commented:
@rtman can you also take care of this one and remove the massive, redundant heading on mobile? 🤦‍♂ Bonus points for fixing that black backgrounded text.

![IMG_8033](https://user-images.githubusercontent.com/273937/61768231-aa49a500-ad9b-11e9-9aa1-505dc6801a71.PNG)
![IMG_8034](https://user-images.githubusercontent.com/273937/61768233-aa49a500-ad9b-11e9-9636-b7eb562954aa.PNG)

micahalcorn commented:
Review pagination is another case of where the spinner ends up on top.

![IMG_8035](https://user-images.githubusercontent.com/273937/61768355-13311d00-ad9c-11e9-8a73-1298cdc1fc7e.PNG)

rtman commented:
Ok I'll have a look soon, one option for the overlaying issue is to put a solid background behind the spinner. So that at least the spinner doesn't appear to mix with the content beneath. But that might not work with all usages or the UI design.

rtman commented:
@micahalcorn I fixed up that notification loading indicator and the black background on the tx hashes. Also took a stab at some styling for the spinner. I added a raised white background to it, looks better imo but let me know what you think/want to do. Here's an updated [test build](https://originprotocol.github.io/test-builds/loading-to-spinner/#/).

micahalcorn commented:
Thanks @rtman. Does it work without the shadow? Most of our designs are flat, and the shadow unnecessarily complicates the predominate case where the space behind the spinner is empty. ⬜️

rtman commented:
@micahalcorn yup, I've removed the shadow and here is the updated [test build](https://originprotocol.github.io/test-builds/loading-to-spinner/#/)

micahalcorn commented:
@rtman one last thing 😇 Maybe some translucency? How about `opacity: 0.8;`? What do you think?

rtman commented:
@micahalcorn I think it looks pretty good with opacity. It's an easy fix if we decide to go back. Want me to commit this?
![opacity_spinner](https://user-images.githubusercontent.com/20849135/61850887-3e823d80-ae6a-11e9-947b-c3f914f7fde3.jpg)

micahalcorn commented:
![ship](https://user-images.githubusercontent.com/273937/61851093-c9633800-ae6a-11e9-950c-8ac0a99173cd.gif)

rtman commented:
@micahalcorn  haha, how about 0.9 opacity? Looks a bit better to me, but up to you!
![opacity_09](https://user-images.githubusercontent.com/20849135/61851117-e7309d00-ae6a-11e9-8b63-2b0f9df35b64.jpg)

rtman commented:
@micahalcorn ok I've updated the opacity and padding on the spinner, [test build](https://originprotocol.github.io/test-builds/loading-to-spinner/#/) is updated if you want to have a look.
